### PR TITLE
Pickle logging tests

### LINKED ISSUE #1310: Test pickling and logging of bond, angle, ... potentials
## Description

We currently do not test the logging or pickling capacities of almost all forces outside pair forces.

## Proposed solution

Check expected loggables, and picklability for 
- bonds
- angles
- dihedrals
- impropers
- special pairs
- external
potentials.

## Additional context

`check_logging` and `check_pickling` in `hoomd/conftest.py` should make this task relatively straightforward.

### PR DESCRIPTION
**Pull request by kodyt**
## Description

Addresses #1310 by adding basic `test_logging` and `test_pickling` functions to the unaccounted forces.
## Motivation and context

Resolves #1310 
 
## How has this been tested?

This is the test

## Change log

```
Adds test_logging and test_pickling functions to test_bond.py, test_angle.py, test_dihedral.py, 
test_improper.py, test_special_pair.py, test_external.py, and test_wall_potential.py. 
```

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/CONTRIBUTING.md).
- [x] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/ContributorAgreement.md).
- [x] My name is on the list of contributors (`sphinx-doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

**Review by syjlee — APPROVED** [review]
Looks good to me.

File: hoomd/md/pytest/test_angle.py [hunk 1]
```diff
@@ -167,3 +171,60 @@ def test_forces_and_energies(triplet_snapshot_factory, simulation_factory,
...
+                                     'default': True
+                                 },
+                                 'additional_energy': {
+                                     'category': LoggerCategories.scalar,
+                                     'default': True
+                                 },
+                                 'additional_virial': {
+                                     'category': LoggerCategories.sequence,
+                                     'default': True
+                                 }
```
Comment (joaander) [inline hoomd/md/pytest/test_angle.py]: This dictionary is repeated for all the Force logging checks. Store this in `conftest.py` and reuse the variable in all locations to reduce code duplication.

**Review by joaander — CHANGES_REQUESTED** [review]
Thanks! This is clean and complete. Thanks for also configuring your git identity correctly.

I have one suggestion for improvement (see the inline comment below).

File: hoomd/md/pytest/test_angle.py (hunk 1 above)
Comment (kodyt) [reply hoomd/md/pytest/test_angle.py]: That's a good idea, I will fix that.

**Review by joaander — APPROVED** [review]
Thanks!

**Pull request merged**